### PR TITLE
Fix/50: Follow semver for 0.x releases

### LINKED ISSUE #50: Support breaking changes in `0.x` bumping minor version

### PR DESCRIPTION
**Pull request by EdgyEdgemond**
closes #50

### CODE REVIEW

File: changelog_gen/extractor.py [hunk 1]
```diff
@@ -82,13 +82,22 @@ def extract_version_tag(sections: SectionDict) -> str:
     Bugs/Fixes: patch
 
     """
+    version_info_ = BumpVersion.get_version_info("patch")
```
Comment (EdgyEdgemond) [inline changelog_gen/extractor.py:85]: explicit tests for this function have been added in a later PR, so currently tested implictly by higher up tests.

File: changelog_gen/extractor.py [hunk 2]
```diff
@@ -82,13 +82,22 @@ def extract_version_tag(sections: SectionDict) -> str:
...
     for section_issues in sections.values():
         for issue in section_issues.values():
             if issue["breaking"]:
                 semver = "major"
+
+    if current.startswith("0."):
+        # If currently on 0.X releases, downgrade semver by one, major -> minor etc.
+        idx = semvers.index(semver)
+        semver = semvers[max(idx - 1, 0)]
+
```
Comment (myddrin) [inline changelog_gen/extractor.py:100]: 🔧  would it be simpler to deal with index the whole time?
```
semvers = ["patch", "minor", "major"]
ver_idx = 0  # start as patch
if "feat" in sections:
  ver_idx = 1 

for section_issues in sections.values():
  for issue in section_issues.values():
    if issue["breaking"]:
      ver_idx = len(semvers) - 1

if current.startwith("0."):
  ver_idx = max(ver_idx - 1, 0)

semver = semvers[ver_idx]
```

**Review by myddrin — COMMENTED** [review]
Do we want this as a normal behaviour or with a configuration switch? How do we move from `0.3.0` to `1.0.0` if we keep being bumped down?

Also how do we handled normal behaviour until 1.0.0 is released? (i.e. 0.1 -> 0.2 o x.feat)?

**Comment by EdgyEdgemond** [comment]
To do a 1.0.0 release youd do an explicit `--version-tag=1.0.0` changelog

File: changelog_gen/extractor.py (hunk 2 above)
Comment (EdgyEdgemond) [reply changelog_gen/extractor.py:100]: this gets rearranged a little in a later PR, with simplfication of configuration keys. Will take a look at this in the context of that change.

Comment (EdgyEdgemond) [reply changelog_gen/extractor.py:100]: the later change makes this idea less clean, as we lose the ability to do `if feat in sections` partly due to using dataclasses, and due to support for custom sections, minor could be `feature`, or multiple types.

Comment (EdgyEdgemond) [reply changelog_gen/extractor.py:100]: https://github.com/EdgyEdgemond/changelog-gen/pull/72/files#diff-5ec09b87a087c81907fc97531029c0a529055bc916aa368dacc36cfb429fe67aR118

**Comment by EdgyEdgemond** [comment]
https://semver.org/#spec-item-4

**Pull request merged**